homeassistant/components/met_eireann/weather.py

```diff
@@ -0,0 +1,150 @@
+    )
+
+
+class MetEireannWeather(CoordinatorEntity, WeatherEntity):
+    """Implementation of a Met Éireann weather condition."""
+
+    def __init__(self, coordinator, config, is_metric, hourly):
+        """Initialise the platform with a data instance and site."""
+        super().__init__(coordinator)
+        self._config = config
+        self._is_metric = is_metric
+        self._hourly = hourly
+        self._name_appendix = "-hourly" if hourly else ""
+
+    @property
+    def unique_id(self):
+        """Return unique ID."""
+        return f"{self._config[CONF_LATITUDE]}-{self._config[CONF_LONGITUDE]}{self._name_appendix}"
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        name = self._config.get(CONF_NAME)
+
+        if name is not None:
+            return f"{name}{self._name_appendix}"
+
+        return f"{DEFAULT_NAME}{self._name_appendix}"
+
+    @property
+    def condition(self):
+        """Return the current condition."""
+        return format_condition(
+            self.coordinator.data.current_weather_data.get("condition")
+        )
+
+    @property
+    def temperature(self):
+        """Return the temperature."""
+        return self.coordinator.data.current_weather_data.get("temperature")
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def pressure(self):
+        """Return the pressure."""
+        pressure_hpa = self.coordinator.data.current_weather_data.get("pressure")
+        if self._is_metric or pressure_hpa is None:
+            return pressure_hpa
+
+        return round(convert_pressure(pressure_hpa, PRESSURE_HPA, PRESSURE_INHG), 2)
+
+    @property
+    def humidity(self):
+        """Return the humidity."""
+        return self.coordinator.data.current_weather_data.get("humidity")
+
+    @property
+    def wind_speed(self):
+        """Return the wind speed."""
+        speed_m_s = self.coordinator.data.current_weather_data.get("wind_speed")
+        if self._is_metric or speed_m_s is None:
+            return speed_m_s
+
+        speed_mi_s = convert_distance(speed_m_s, LENGTH_METERS, LENGTH_MILES)
+        speed_mi_h = speed_mi_s / 3600.0
+        return int(round(speed_mi_h))
+
+    @property
+    def wind_bearing(self):
+        """Return the wind direction."""
+        return self.coordinator.data.current_weather_data.get("wind_bearing")
+
+    @property
+    def attribution(self):
+        """Return the attribution."""
+        return ATTRIBUTION
+
+    @property
+    def forecast(self):
+        """Return the forecast array."""
+        if self._hourly:
+            # Format the condition for hourly forecast items
+            for item in self.coordinator.data.hourly_forecast:
+                if "condition" in item:
+                    item["condition"] = format_condition(item["condition"])
+            return self.coordinator.data.hourly_forecast
```

Comment:
Please create the forecast data explicitly. It's a Home Assistant api so we need to do that in the integration.

https://developers.home-assistant.io/docs/core/entity/weather#forecast